gdc_client/parcel/segment.py

```diff
@@ -55,24 +57,24 @@ def __init__(self, download, n_procs):
         self.download = download
         self.n_procs = n_procs
         self.pbar = None
+        self.done = False
 
         # Initialize producer
         self.load_state()
-        self._setup_pbar()
-        self._setup_queues()
-        self._setup_work()
-        self.schedule()
+        if not self.done:
+            self._setup_pbar()
+            self._setup_queues()
+            self._setup_work()
+            self.schedule()
 
     def _setup_pbar(self):
         self.pbar = get_file_transfer_pbar(self.download.url, self.download.size)
 
     def _setup_work(self):
-        if self.is_complete():
-            log.debug("File already complete.")
-            return
-
         work_size = self.integrate(self.work_pool)
         self.block_size = work_size // self.n_procs
+        self.total_tasks = math.ceil(work_size / self.block_size)
+        log.debug("Total number of tasks: {0}".format(self.total_tasks))
```

Comment:
To add to the confusion loggers have their own string formatting syntax that resembles C's `printf` function. Linters will raise a warning if loggers don't follow this format, too.

```python
log.debug("Total number of tasks: %s", self.total_tasks)
```

It's not exactly the same as `printf` because `%s` can be used with any Object, and it's `__str__()` method gets called to format the string.